src/utils/handlers/general/searchTrack.ts

```diff
@@ -98,7 +98,10 @@ export async function searchTrack(
                     }
 
                     case "playlist": {
-                        const playlist = await youtube.getPlaylist(url.toString());
+                        const list = /list=([^&]+)/g.exec(url.href) ?? [];
+                        const playlist = await youtube.getPlaylist(list[1].toString());
+                        const currentSong = /index=(\d+)/.exec(url.href);
+                        const songIndex = currentSong ? currentSong[1] : undefined;
```

Comment:
Use `url.searchParams.get` to get value of `list` and `index` query.

```suggestion
                        const list = url.searchParams.get("list") ?? "";
                        const playlist = await youtube.getPlaylist(list);
                        const songIndex = url.searchParams.get("index");
```